apps/opik-backend/src/main/java/com/comet/opik/domain/TraceDAO.java

```diff
@@ -57,13 +61,49 @@ Mono<Void> partialInsert(UUID projectId, TraceUpdate traceUpdate, UUID traceId,
 
     Mono<List<WorkspaceAndResourceId>> getTraceWorkspace(Set<UUID> traceIds, Connection connection);
 
+    Mono<Long> batchInsert(List<Trace> traces, Connection connection);
 }
 
 @Slf4j
 @Singleton
 @RequiredArgsConstructor(onConstructor_ = @Inject)
 class TraceDAOImpl implements TraceDAO {
 
+    private static final String BATCH_INSERT = """
+            INSERT INTO traces(
+                id,
+                project_id,
+                workspace_id,
+                name,
+                start_time,
+                end_time,
+                input,
+                output,
+                metadata,
+                tags,
+                created_by,
+                last_updated_by
+            ) VALUES
+                <items:{item |
+                    (
+                        :id<item.index>,
+                        :project_id<item.index>,
+                        :workspace_id,
+                        :name<item.index>,
+                        parseDateTime64BestEffort(:start_time<item.index>, 9),
+                        if(:end_time<item.index> IS NULL, NULL, parseDateTime64BestEffort(:end_time<item.index>, 9)),
```

Comment:
Minor: `parseDateTimeBestEffortUSOrNull` can be used:
- https://clickhouse.com/docs/en/sql-reference/functions/type-conversion-functions#parsedatetimebesteffortusornull